File: mobile-wallet-master/mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/LinkBankAccountPresenter.java
package org.mifos.mobilewallet.mifospay.bank.presenter;

import org.mifos.mobilewallet.core.domain.model.BankAccountDetails;
import org.mifos.mobilewallet.mifospay.bank.BankContract;
import org.mifos.mobilewallet.mifospay.base.BaseView;

import java.util.Random;

import javax.inject.Inject;

/**
 * Created by ankur on 09/July/2018
 */

public class LinkBankAccountPresenter implements BankContract.LinkBankAccountPresenter {

    BankContract.LinkBankAccountView mLinkBankAccountView;

    @Inject
    public LinkBankAccountPresenter() {

    }

    @Override
    public void attachView(BaseView baseView) {
        mLinkBankAccountView = (BankContract.LinkBankAccountView) baseView;
        mLinkBankAccountView.setPresenter(this);
    }

    @Override
    public void fetchBankAccountDetails(final String bankName) {
        // TODO:: UPI API implement
        mLinkBankAccountView.addBankAccount(
                new BankAccountDetails(bankName, "Ankur Sharma", "New Delhi",
                        new Random().nextInt() + " ", "Savings"));
    }
}

File: mifospay/src/main/java/org/mifospay/bank/presenter/LinkBankAccountPresenter.kt
package org.mifospay.bank.presenter

import com.mifospay.core.model.domain.BankAccountDetails
import org.mifospay.bank.BankContract
import org.mifospay.bank.BankContract.LinkBankAccountView
import org.mifospay.base.BaseView
import java.util.Random
import javax.inject.Inject

/**
 * Created by ankur on 09/July/2018
 */
class LinkBankAccountPresenter @Inject constructor() : BankContract.LinkBankAccountPresenter {
    var mLinkBankAccountView: LinkBankAccountView? = null
    override fun attachView(baseView: BaseView<*>?) {
        mLinkBankAccountView = baseView as LinkBankAccountView?
        mLinkBankAccountView!!.setPresenter(this)
    }

    override fun fetchBankAccountDetails(bankName: String?) {
        // TODO:: UPI API implement
        mLinkBankAccountView!!.addBankAccount(
            BankAccountDetails(
                bankName, "Ankur Sharma", "New Delhi",
                mRandom.nextInt().toString() + " ", "Savings"
            )
        )
    }

    companion object {
        private val mRandom = Random()
    }
}